ios/PurchasesHybridCommon/PurchasesHybridCommon/CommonFunctionality.swift

```diff
@@ -350,18 +319,30 @@ import RevenueCat
     @objc(makeDeferredPurchase:completionBlock:)
     static func makeDeferredPurchase(_ startPurchase: StartPurchaseBlock,
                                      completion: @escaping ([String: Any]?, ErrorContainer?) -> Void) {
-        startPurchase { transaction, customerInfo, error, userCancelled in
+        startPurchase(Self.createPurchaseCompletionBlock(completion: completion))
+    }
+
+    private static func createPurchaseCompletionBlock(
+        completion: @escaping ([String: Any]?, ErrorContainer?) -> Void
+    ) -> @Sendable (StoreTransaction?,
+                    CustomerInfo?,
+                    Error?,
+                    Bool) -> Void {
+        return { transaction, customerInfo, error, userCancelled in
             if let error = error {
                 completion(nil, Self.createErrorContainer(error: error, userCancelled: userCancelled))
             } else if let customerInfo = customerInfo,
                       let transaction = transaction {
                 completion([
                     "customerInfo": customerInfo.dictionary,
-                    "productIdentifier": transaction.productIdentifier
+                    "productIdentifier": transaction.productIdentifier,
+                    "transaction": transaction.dictionary
```

Comment:
We probably need to do this as well for android, but that can come in a different PR.

Reply:
#686